### DISCUSSION TITLE
Timeline in incorrect order within a block

### DISCUSSION THREAD

**Discussion by tprus-luc** [CollectionBuilder/collectionbuilder.github.io#138] [Q&A] [answered]
It appears that on the Timeline page, within a block, such as a year, items are displayed in semi-random order. Items with a shared date are generally together, but then they may be followed by an item that should be displayed above that group, as it has an earlier date.

Is it possible to prevent this behavior?

#99 addressed the Browse page's randomization by editing JavaScript, but the Timeline's equivalent (_includes/js/timeline-js.html) does not have equivalent code. Where is the code for this behavior located, or is there some other way to prevent this?

**Comment by evanwill** [comment]
@tprus-luc the Timeline page is created all with Liquid (not javascript in this case). You can find how things come together by starting at the page stub and looking at the layouts it calls. 
For timeline, "pages/timeline.md" has layout `timeline` , so you look in "_layouts/timeline.html". 
The Liquid code at the top figures out all the unique years in your metadata, then iterates over those years to create each block. 
Each block grabs the related items using the Liquid `{%- assign inYear = items | where_exp: 'item', 'item[field] contains year' -%}`, then iterates over that set to put the items in the year block. The order of items in this case will be what ever the order is in the spreadsheet (not randomized). Liquid isn't great a sorting, but if your "date" field was all consistent ISO, you could add a sort to that expression to get them to sort in date order. Replace the expression above with:
`{%- assign inYear = items | where_exp: 'item', 'item[field] contains year' | sort: 'date' -%}`

**Reply by tprus-luc** [reply]
Are the items sorted by objectid rather than date on the Timeline page?

I have quite a few items that are in consecutively numbered in the `objectid` column (let's say 12 through 40) and all have a common date of October 19, 2024. Those items are grouped together on the Timeline page inside the 2024 block.

There is an item with the date of May 19, 2024 and the objectid of 59. On the timeline, within the 2024 block, this object displays after the October 19 items despite the earlier date!

Also, the item with the objectid 265 but a date of October 19, 2024 displays after all of those I have just listed, that is, after object 59, again despite the date.

I tried the change in "_layouts/timeline.html" but I get this error in the console window:

>   Liquid Exception: Liquid error (line 47): comparison of Hash with Hash failed in
 [path to my project]/_layouts/timeline.html
>             Error: Liquid error (line 47): comparison of Hash with Hash failed
>            Error: Run jekyll build --trace for more information.

All dates are in consistent ISO-8601 format.

**Reply by evanwill** [reply, marked as answer]
@tprus-luc sorry, on the "sort" I left out the field to add, it should look like: 
`{%- assign inYear = items | where_exp: 'item', 'item[field] contains year' | sort: 'date' -%}`

adding that will actually sort them based on your date field (keep in mind, it is not parsing the dates, just sorting them alphabetically, which will work if your dates are yyyy-mm-dd, but won't work if they are say yyyy-m-d or d-m-yyyy). 
By default, it is not sorting by objectid, it is just the order in the CSV (so more or less sorting by the index number in the spreadsheet).

**Reply by tprus-luc** [reply]
Thank you! Please make sorting by the date field the default, it is what one would expect on a page named Timeline.

**Reply by evanwill** [reply]
I did add that in CB-CSV at this point, https://github.com/CollectionBuilder/collectionbuilder-csv/commit/ff07c1bfa5087bc2221aee815c1cb94591b60fef
